### PR TITLE
Enable attach message reordering - next

### PR DESCRIPTION
**Pull request by andre4i**
This feature was initially planned to be shipped as enabled but it ended up being disabled.

This feature allows for attach messages to be submitted as a separate batch. See https://github.com/microsoft/FluidFramework/pull/12030 for details

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts [hunk 1]
```diff
@@ -43,6 +43,8 @@ describeFullCompat("GC Data Store Aliased", (getTestObjectProvider) => {
         const ds1 = await requestFluidObject<ITestDataObject>(aliasableDataStore1, "");
 
         (aliasableDataStore1 as any).fluidDataStoreChannel.bindToContext();
+        // yield a turn to ensure we get to flush the attach op
+        await new Promise<void>((resolve) => setTimeout(resolve));
```
Comment (vladsud) [inline packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts:47]: Curious, why is this required? I.e. how change in behavior results in this change? Ops are flushed async either way, so I'd expect not requiring this change?
This suggest that change is breaking, i.e. it cases change in behavior that clients might need to react to.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -2760,28 +2762,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                             });
                     }
                 }
-            } else {
```
Comment (vladsud) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2763]: Given that we had these fights in bigger Office for the long time, I think it would be great to learn from the past.
I feel like there are two ways to approach formatting:
1. We agree on specific formatting and encode it into build (linting / prettier) - having wrong format is a build break.
2. We adopt the strategy - change formatting only for code that you are touching.
   - Scope is important here. File might be that scope. But I, personally, find it difficult - I'd prefer changes to only represent "real" changes.

Thoughts?

I'd personally prefer # 1 - it's so much easier to work in Loop repo because of it - I do not even need to think about order of imports or any other stuff - tools do it for me.

+ @tylerbutler for his thoughts, as I think this topic come up many times.

File: packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts (hunk 1 above)
Comment (andre4i) [reply packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts:47]: This is not required. It's actually overkill. Will remove it.

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 2 above)
Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2763]: I prefer 1 as well.

File: packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts (hunk 1 above)
Comment (andre4i) [reply packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts]: Leftover from testing, sorry..

**Review by NicholasCouri — APPROVED** [review]
